AWSClientRuntime/Sources/Auth/AWSCredentialsProvider.swift

```diff
@@ -6,89 +6,123 @@ import AwsCommonRuntimeKit
 import ClientRuntime
 
 public class AWSCredentialsProvider: CredentialsProvider {
-    let crtCredentialsProvider: CRTAWSCredentialsProvider
-    private let sharedDefaultIO: SDKDefaultIO = SDKDefaultIO.shared
+    let crtCredentialsProvider: AwsCommonRuntimeKit.CredentialsProvider
     
-    init(awsCredentialsProvider: CRTAWSCredentialsProvider) {
+    init(awsCredentialsProvider: AwsCommonRuntimeKit.CredentialsProvider) {
         self.crtCredentialsProvider = awsCredentialsProvider
     }
     
     public static func fromEnv() throws -> AWSCredentialsProvider {
```

Comment:
all these static methods look scary from testing point of view. Not related to this PR. I would much prefer a singleton pattern.

Reply:
Are we only ever supposed to have a single credentials provider instance (for each type?), if not then im not sure we want singleton pattern here. (ideally i'd like to avoid singletons and global state whenever possible). 

But I agree, the current interface for creating credential providers has much room for improvement